vips/image.go

```diff
@@ -654,20 +655,22 @@ func (r *ImageRef) Invert() error {
 	return nil
 }
 
-// Resize executes the 'resize' operation
 func (r *ImageRef) Resize(scale float64, kernel Kernel) error {
-	err := r.PremultiplyAlpha()
-	if err != nil {
-		return err
+	var out *C.VipsImage
+	var err error
+
+	if scale < 1 && r.HasAlpha() {
+		out, err = vipsAlphaResize(r.image, scale)
+	} else {
+		out, err = vipsResize(r.image, scale, kernel)
```

Comment:
I think we might be missing the PremultiplyAlpha() when scale > 1, but I might be wrong...

Reply:
aaah you're right, I'll do that. Good thing we have code review.
Just to make sure, this scenario doesn't actually exist in image manipulator right?

Reply:
scale > 1? it exist, we don't prohibit upscaling and in some scenarios do them intently (as a fallback of superscale)

Reply:
So we might still get the alpha artifacts when upscaling

Reply:
Upscaling alpha containing images is allowed